amm/traits/stable_pool.rs

```diff
@@ -23,45 +23,49 @@ pub trait StablePoolView {
     #[ink(message)]
     fn fees(&self) -> (u32, u32);
 
+    /// Protocol fees receiver (if any)
+    #[ink(message)]
+    fn fee_receiver(&self) -> Option<AccountId>;
+
     /// Updates cached token rates if expired and
     /// returns current tokens rates with precision of 12 decimal places.
     #[ink(message)]
     fn token_rates(&mut self) -> Vec<u128>;
 
-    /// Calculate swap amount of token_out
-    /// given token_in amount.
-    /// Returns (amount_out, fee)
-    /// NOTE: fee is applied to token_out
+    /// Calculate swap amount of `token_out`
+    /// given `token_in amount`.
+    /// Returns a tuple of (amount out, fee)
+    /// NOTE: fee is applied on `token_out`
     #[ink(message)]
     fn get_swap_amount_out(
-        &mut self,
+        &self,
         token_in: AccountId,
         token_out: AccountId,
         token_in_amount: u128,
     ) -> Result<(u128, u128), StablePoolError>;
 
-    /// Calculate required amount of token_in
-    /// given swap token_out amount
-    /// Returns (amount_in, fee)
-    /// fee is applied to token_out
+    /// Calculate required swap amount of `token_in`
+    /// to get `token_out_amount`.
+    /// Returns a tuple of (amount in, fee)
+    /// NOTE: fee is applied on `token_out`
     #[ink(message)]
     fn get_swap_amount_in(
-        &mut self,
+        &self,
         token_in: AccountId,
         token_out: AccountId,
         token_out_amount: u128,
     ) -> Result<(u128, u128), StablePoolError>;
 
     /// Calculate how many lp tokens will be minted
     /// given deposit `amounts`.
-    /// Returns (lp_amount, fee)
+    /// Returns a tuple of (lpt amount, fee)
     #[ink(message)]
     fn get_mint_liquidity_for_amounts(
-        &mut self,
+        &self,
         amounts: Vec<u128>,
     ) -> Result<(u128, u128), StablePoolError>;
 
-    /// Calculate ideal deposit amounts required
+    /// Estimate ideal deposit amounts required
```

Comment:
I would stay with 'Calculate' here and below. 'Estimate' suggests some randomness :stuck_out_tongue:. I suppose that 'Estimate' here was supposed to account for precision errors. I think 'Calculate' is still a good word, even in presence of precision errors.